include/ostatus.php

```diff
@@ -353,17 +353,25 @@ public static function import($xml,$importer,&$contact, &$hub) {
 				continue;
 			}
 
-			$item["body"] = add_page_info_to_body(html2bbcode($xpath->query('atom:content/text()', $entry)->item(0)->nodeValue));
 			$item["object-type"] = $xpath->query('activity:object-type/text()', $entry)->item(0)->nodeValue;
+			$item["verb"] = $xpath->query('activity:verb/text()', $entry)->item(0)->nodeValue;
+
+			if ($item["verb"] == ACTIVITY_POST && $xpath->evaluate('boolean(atom:summary)', $entry)) {
+				// Mastodon Content Warning
+				$clear_text = $xpath->query('atom:summary/text()', $entry)->item(0)->nodeValue;
+				$hidden_text = $xpath->query('atom:content/text()', $entry)->item(0)->nodeValue;
 
-			if (($item["object-type"] == ACTIVITY_OBJ_BOOKMARK) OR ($item["object-type"] == ACTIVITY_OBJ_EVENT)) {
+				$item["body"] = add_page_info_to_body(html2bbcode($clear_text) . '[spoiler]' . html2bbcode($hidden_text) . '[/spoiler]');
```

Comment:
Maybe it would be wiser to use "add_page_info_to_body" on the hidden text instead of the summary.